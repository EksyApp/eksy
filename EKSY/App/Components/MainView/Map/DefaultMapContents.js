import React, {Component} from 'react'
import Markers from './Markers'
import {View} from 'react-native'
import UserCircle from './UserCircle'
import {LocationShape, MarkersShape, RegionShape} from '../../../Utils/PropTypeShapes'
import PropTypes from 'prop-types'

export default class DefaultMapContents extends Component {
	
	render() {
		return (
				<View>
					<Markers
							markerList={this.props.markerList}
							onMarkerClick={this.props.onMarkerClick}
					/>
					<UserCircle
							currentLocation={this.props.currentLocation}
							radius={this.props.radius}
							currentRegion={this.props.currentRegion}
					/>
				</View>
		)
	}
	
}

DefaultMapContents.propTypes = {
	markerList: MarkersShape,
	currentLocation: LocationShape,
	radius: PropTypes.number,
	currentRegion: RegionShape,
	onMarkerClick: PropTypes.func
}