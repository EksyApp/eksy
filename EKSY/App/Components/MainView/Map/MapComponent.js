import React, {Component} from 'react'
import {
	StyleSheet,
	View,
	Animated,
	Dimensions
} from 'react-native'
import MapView from 'react-native-maps'
import MapManager from '../../../Utils/MapManager'
import Marker from './Marker'
import PropTypes from 'prop-types'
import {LocationShape, MarkerShape, MarkersShape, RegionShape, RouteShape} from '../../../Utils/PropTypeShapes'
import GeoFire from 'geofire'

const Screen = {
	width: Dimensions.get('window').width,
	height: Dimensions.get('window').height - 75
}

//Renders map view inside the main view (MainViewComponent)
//Map component holds clickable markers (Marker)
export default class MapComponent extends Component {
	
	constructor(props) {
		super(props)
		this._manager = new MapManager()
		this._manager.setMapObject(this)
		this._map = null
		
		this.state = {
			zoomLevel: (360 * ((Screen.width / 256) / this.props.currentRegion.longitudeDelta)) + 1
		}
	}
	
	animateToCoordinate(position, delay) {
		if (this._map) {
			this._map._component.animateToCoordinate(position, delay)
		}
	}
	
	handleRegionChange(region) {
		this.props.regionChange(region)
		this.setState({zoomLevel: (360 * ((Screen.width / 256) / region.longitudeDelta)) + 1})
	}
	
	renderMarkers() {
		return this.props.markerList.map((marker, index) =>
				<Marker
						data={marker}
						setMarkerSelected={this.props.setMarkerSelected}
						setMarkerViewVisible={this.props.setMarkerViewVisible}
						disableGestures={this.props.disableGestures}
						key={marker.key}
				/>
		)
		
	}
	
	
	renderUserCircle() {
		if (this.props.currentLocation.isKnown) {
			return (
					<Animated.View>
						<MapView.Circle center={this.props.currentLocation}
						                radius={this.props.radius * 1000}
						                strokeWidth={0.5}
						                strokeColor="rgba(66, 180, 230, 1)"
						                fillColor="rgba(66, 180, 230, 0.2)"
						/>
						<MapView.Circle center={this.props.currentLocation}
						                radius={300000 / this.state.zoomLevel}
						                strokeWidth={0.5}
						                strokeColor="rgba(66, 180, 230, 1)"
						                fillColor="rgba(66, 180, 230, 1)"
						/>
					</Animated.View>
			)
		}
		
		return null
	}
	
	renderRouteLine() {
		if (this.props.routeIsActive) {
			let distance = GeoFire.distance([this.props.currentLocation.latitude, this.props.currentLocation.longitude], [this.props.nextMarker.latitude, this.props.nextMarker.longitude])
			console.log(distance)
			console.log(this.props.radius)
			if (distance >= this.props.radius) {
				let latitude = this.props.currentLocation.latitude + (this.props.radius / distance) * (this.props.nextMarker.latitude - this.props.currentLocation.latitude)
				let longitude = this.props.currentLocation.longitude + (this.props.radius / distance) * (this.props.nextMarker.longitude - this.props.currentLocation.longitude)
				return (<MapView.Polyline coordinates={[this.props.currentLocation, {latitude, longitude}]}/>)
			} else {
				return (<MapView.Polyline coordinates={[this.props.currentLocation, this.props.nextMarker]}/>)
			}
		}
	}
	
	renderMapView() {
		return (
				<MapView.Animated
						ref={(ref) => this._map = ref}
						style={styles.map}
						initialRegion={this.props.currentRegion}
						onRegionChange={(region) => this.handleRegionChange(region)}
						followUserLocation
						showsMyLocationButton={true}
						showsBuildings={true}
						showCompass={false}
				>
					{this.renderRouteLine()}
					{this.renderUserCircle()}
					{this.renderMarkers()}
				</MapView.Animated>
		)
	}
	
	render() {
		return (
				<View style={styles.container}>
					{this.renderMapView()}
				</View>
		)
	}
	
	
}


const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: 'flex-end',
		alignItems: 'center',
	},
	map: {
		position: 'absolute',
		top: 0,
		bottom: 0,
		left: 0,
		right: 0,
	},
	buttonContainer: {
		marginVertical: 20,
	},
	button: {
		alignItems: 'center',
		backgroundColor: 'rgba(255,255,255,0.7)',
		borderRadius: 20,
		padding: 12,
		width: 160,
	},
	callout: {
		width: 140,
	},
	circle: {
		borderRadius: 12,
		backgroundColor: 'rgba(130,4,150, 0.3)',
		position: 'absolute',
		borderWidth: 1,
		borderColor: 'rgba(130,4,150, 0.5)'
	}
})

MapComponent.propTypes = {
	currentRegion: RegionShape,
	currentLocation: LocationShape,
	radius: PropTypes.number,
	markerList: MarkersShape,
	setMarkerSelected: PropTypes.func,
	setMarkerViewVisible: PropTypes.func,
	disableGestures: PropTypes.func,
	routeIsActive: PropTypes.bool,
	route: RouteShape,
	nextMarker: MarkerShape
}

