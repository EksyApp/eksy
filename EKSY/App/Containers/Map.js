import React, { Component } from 'react';
import {
  StyleSheet,         // CSS-like styles
  Text,               // Renders text
  TouchableOpacity,   // Pressable container
  View                // Container component
} from 'react-native';
import MapView from 'react-native-maps';
// Import data
import { characters } from './data';
import Callout from './Callout';
import styles from './Styles/MapStyles'

export default class Map extends Component {

  state = {
    // Show good or all characters flag
    showGoodOnly: false,
  }

  render() {
    return (
      <View style={styles.container}>
      {/* Map*/}
      <MapView
      style={styles.map}
      // Position on Manhattan, New York
      initialRegion={{
        latitude: 40.77096,
        longitude: -73.97702,
        latitudeDelta: 0.0491,
        longitudeDelta: 0.0375,
      }}
      >
      {/* Loop through characters and add pins on the map */}
      {characters.map((character, index) =>
            // If showGoodOnly is true, but the character is bad - do not show it
            this.state.showGoodOnly && !character.good || <MapView.Marker
              coordinate={{
                latitude: character.coordinate[0],
                longitude: character.coordinate[1],
              }}
              // Callout offset
              calloutOffset={{ x: -8, y: 28 }}
              // Greed color for good characters and red for others
              pinColor={character.good ? '#009688' : '#f44336'}
              key={index}
            >
              {/* Callout */}
              <MapView.Callout tooltip style={styles.callout}>
                <Callout
                  name={character.name}
                  image={character.image}
                />
              </MapView.Callout>
            </MapView.Marker>
          )}
      </MapView>
      {/* Button */}
      <View style={styles.buttonContainer}>
      <TouchableOpacity
      style={styles.button}
      // Toggle this.state.showGoodOnly
      onPress={() => this.setState({
        showGoodOnly: !this.state.showGoodOnly
      })}
      >
      <Text>{this.state.showGoodOnly ? 'Show All' : 'Show Good Only'}</Text>
      </TouchableOpacity>
      </View>
      </View>
    );
  }
}
