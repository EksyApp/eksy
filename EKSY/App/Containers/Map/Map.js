import React, {Component} from 'react'
import {
	StyleSheet,         // CSS-like styles
	Text,               // Renders text
	TouchableOpacity,   // Pressable container
	View,               // Container component
	ActivityIndicator,
	Animated,
	Dimensions
} from 'react-native'
import MapView from 'react-native-maps'
import MapManager from './MapManager'
import Marker from './Marker'
import isEqual from 'lodash/isEqual'
import {connect} from 'react-redux'

const Screen = {
	width: Dimensions.get('window').width,
	height: Dimensions.get('window').height - 75
}

export default class Map extends Component {
	
	constructor(props) {
		super(props)
		this._manager = new MapManager();
		this._manager.setMapObject(this);
		this._map = null;
		
		this.state = {
			zoomLevel: (360 * ((Screen.width/256) / this.props.currentRegion.longitudeDelta)) + 1
		}
	}
	
	animateToCoordinate(position, delay) {
		if (this._map) {
			this._map.animateToCoordinate(position, delay)
		}
	}
	
	handleRegionChange(region) {
		this.props.regionChange(region)
		this.setState({...this.state, zoomLevel: (360 * ((Screen.width/256) / region.longitudeDelta)) + 1})
		console.log(this.state.zoomLevel)
	}
	
	async componentWillReceiveProps(nextProps) {
		if (!isEqual(nextProps, this.props)) {
			// this.store = await Store()
			// navigator.geolocation.getCurrentPosition(
			// 	(position) => {
			// 		this.store.dispatch(Actions.updateLocation(position.coords))
			// 		this.store.dispatch(Actions.locationKnown(true))
			// 		this.animateToCoordinate(position.coords, 100)
			// 	},
			// 	(error) => this.store.dispatch(Actions.locationKnown(false)),
			// 	{enableHighAccuracy: false, timeout: 500, maximumAge: 1000000, distanceFilter: 1}
			// )
		}
	}
	
	renderMarkers() {
		return this.props.markerList.map((marker, index) =>
				<Marker
						data={marker}
						setMarkerSelected={this.props.setMarkerSelected}
						setMarkerViewVisible={this.props.setMarkerViewVisible}
						disableGestures={this.props.disableGestures}
						key={marker.key}
				/>)
	}
	
	renderUserCircle() {
		if (this.props.currentLocation.isKnown) {
			return (
					<Animated.View>
						<MapView.Circle center={this.props.currentLocation}
						                radius={100}
						                strokeWidth={0.5}
						                strokeColor="rgba(66, 180, 230, 1)"
						                fillColor="rgba(66, 180, 230, 0.2)"
						/>
						<MapView.Circle center={this.props.currentLocation}
						                radius={300000 / this.state.zoomLevel}
						                strokeWidth={0.5}
						                strokeColor="rgba(66, 180, 230, 1)"
						                fillColor="rgba(66, 180, 230, 1)"
						/>
					</Animated.View>
			)
			// <MapView.Marker key='user' coordinate={this.props.currentLocation} style={{flex: 1}}>
			// 	<Animated.View style={[styles.userMarkerWrap]}>
			// 		<Animated.View style={[styles.userRing]} />
			// 		<View style={styles.userMarker} />
			// 	</Animated.View>
			// </MapView.Marker>
			
			// <MapView.Circle
			// 					center={this.props.currentLocation}
			// 					radius={100}
			// 					strokeWidth={2}
			// 					strokeColor={circleStrokeColor}
			// 					fillColor={circleFillColor}
			// 					key={(this.props.currentLocation.longitude + this.props.currentLocation.latitude)}
			// 				/>
			// <MapView.Marker
			// 		key='user'
			// 		coordinate={this.props.currentLocation}
			// 		style={{flex: 1}}
			// >
		}
		
		return null
	}
	
	renderMapView() {
		return (
				<MapView.Animated
						ref={(ref) => this._map = ref}
						style={styles.map}
						initialRegion={this.props.currentRegion}
						onRegionChange={(region) => this.handleRegionChange(region)}
						/*loadingEnabled*/
						/*showsUserLocation*/
						followUserLocation
						showsMyLocationButton={true}
						showsBuildings={true}
						showCompass={false}
				>
					{this.renderUserCircle()}
					{this.renderMarkers()}
				</MapView.Animated>
		)
	}
	
	render() {
		return (
				<View style={styles.container}>
					{this.renderMapView()}
				</View>
		)
	}
}


const styles = StyleSheet.create({
	container: {
		flex: 1,                            // Take up the whole screen
		justifyContent: 'flex-end',         // Arrange button at the bottom
		alignItems: 'center',               // Center button horizontally
	},
	map: {
		position:'absolute',
		top: 0,
		bottom: 0,
		left: 0,
		right: 0,
	},
	buttonContainer: {
		marginVertical: 20,
	},
	button: {
		alignItems: 'center',
		backgroundColor: 'rgba(255,255,255,0.7)',
		borderRadius: 20,
		padding: 12,
		width: 160,
	},
	callout: {
		width: 140,
	},
	circle: {
		borderRadius: 12,
		backgroundColor: "rgba(130,4,150, 0.3)",
		position: "absolute",
		borderWidth: 1,
		borderColor: "rgba(130,4,150, 0.5)"
	}
});