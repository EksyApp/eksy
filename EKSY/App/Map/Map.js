import React, { Component } from 'react'
import {
  StyleSheet,         // CSS-like styles
  Text,               // Renders text
  TouchableOpacity,   // Pressable container
  View,               // Container component
  ActivityIndicator
} from 'react-native'
import MapView from 'react-native-maps'
import Callout from './Callout'
import styles from './Styles/MapStyles'
import MapManager from './MapManager'
import PostOffice from '../lib/PostOffice'
import * as Actions from '../Actions'

import testData from '../includes/data/Sarjakuvat.json'

export default class Map extends Component {

  constructor (props) {
    super(props)
    this._manager = new MapManager();
    this._manager.setMapObject(this);
    this._map = null;
    this.po = new PostOffice();
    this.state = {
      mapLoaded: false
    }

  }

  update () {
    this.forceUpdate()
  }

  componentDidMount () {
    this.setState({ mapLoaded: true })
    this._manager.goToCurrentPosition();
  }

  animateToCoordinate (position, delay) {
    if (this.state.mapLoaded) {
      this._map.animateToCoordinate(position, delay)
    } else {
      console.warn('Map not rendered when used animateToCoordinate')
    }
  }

  renderPampylat () {
     return testData.features.map((pampyla, index) => {
       console.log(pampyla.properties.gx_media_links)
       return (
           // If showGoodOnly is true, but the character is bad - do not show it
         <MapView.Marker
           coordinate={{
             longitude: pampyla.geometry.coordinates[0],
             latitude: pampyla.geometry.coordinates[1]
           }}
           // Callout offset
           calloutOffset={{ x: -8, y: 28 }}
           // Greed color for good characters and red for others
           pinColor={'#009688'}
           key={index}
         >
           {/* Callout */}
           <MapView.Callout tooltip style={styles.callout}>
             <Callout
               name={pampyla.properties.Name}
               description={pampyla.properties.description}
               image={pampyla.properties.gx_media_links}
             />
           </MapView.Callout>
         </MapView.Marker>
       )
     })
   }

   handleRegionChange(region) {
     this.props.regionChange(region)
   }


  renderMapView () {
    if (!this.state.mapLoaded) {
      return (
        <ActivityIndicator size='large' />
      )
    }



    return (
      <MapView
      ref={(ref) => this._map = ref}
      style={styles.map}
      initialRegion={this.props.initialRegion}
      onRegionChangeComplete = {(region) => this.handleRegionChange(region)}
      showsUserLocation
      >
      {this._manager.getMarkers().map((marker, index) => marker.render())}
      {this.renderPampylat()}
      </MapView>
    )
  }

  render () {
    return (
      <View style={styles.container}>
        {this.renderMapView()}
      </View>
    )
  }
}
