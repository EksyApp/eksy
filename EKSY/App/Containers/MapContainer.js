import React, {Component} from 'react'
import {PermissionsAndroid} from 'react-native'
import Map from './Map/Map'
import MenuButton from '../Components/MenuButton'
import {View, StyleSheet, Dimensions, Animated, Text, TouchableWithoutFeedback} from 'react-native'
import {Badge} from 'react-native-elements'
import * as Actions from '../Actions'
import {connect} from 'react-redux'
import Interactable from 'react-native-interactable'
import MarkerCarousel from '../Components/MarkerCarousel'
import {backgroundColor, detailColor} from '../Theme'
import MarkerView from '../Containers/MarkerView'

const Screen = {
	width: Dimensions.get('window').width,
	height: Dimensions.get('window').height - 75
}

export class MapContainer extends Component {

	constructor(props) {
		super(props)

		this._deltaY = new Animated.Value(Screen.height - 100);
	}

	async requestLocationPermission() {
		try {
			const granted = await PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION)
			if (granted !== PermissionsAndroid.RESULTS.GRANTED) {
				console.log('no permissions granted') // prompt user?
			}
		} catch (err) {
			console.warn(err)
		}
	}

	async componentWillMount() {
		await this.requestLocationPermission()
	}

	render() {
		return (
				<View style={styles.viewContainer}>
					<Map
							currentRegion={this.props.currentRegion}
							currentLocation={this.props.currentLocation}
							markerList={this.props.markerList}
							setMarkerSelected={this.props.setMarkerSelected}
							setMarkerViewVisible={this.props.setMarkerViewVisible}
							disableGestures={this.props.disableGestures}
							regionChange={this.props.regionChange}/>
					<MenuButton onPress={() => {
						this.props.menuButtonPress()
					}}/>
					<View style={styles.panelContainer}>
						<Animated.View style={[styles.panelContainer, {
							backgroundColor: 'black',
							opacity: this._deltaY.interpolate({
								inputRange: [0, Screen.height - 100],
								outputRange: [1, 0],
								extrapolateRight: 'clamp'
							})
						}]}
						pointerEvents="none"/>
						<Interactable.View
								verticalOnly={true}
								snapPoints={[{y: Screen.height - 220}, {y: Screen.height}]}
								boundaries={{top: 300}}
								initialPosition={{y: Screen.height}}
								animatedValueY={this._deltaY}>
							<Animated.View style={styles.panel}>
								<Animated.View style={styles.panelHeader}>
									<Animated.View style={styles.panelHandle}/>
									{this.props.markerList.length > 0 &&
									<Badge
											value={this.props.markerList.length}
											containerStyle={styles.badgeContainer}
											textStyle={styles.badgeText}
									/>
									}
								</Animated.View>
								<View style={styles.panelVisible}>
									<MarkerCarousel
											markerList={this.props.markerList}
											setMarkerSelected={this.props.setMarkerSelected}
											setMarkerViewVisible={this.props.setMarkerViewVisible}
											disableGestures={this.props.disableGestures}
											pointerEvents="none"/>
								</View>

							</Animated.View>
						</Interactable.View>
					</View>

					<MarkerView />
				</View>
		)
	}
}

// <View style={styles.panel}>
// </View>
const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	panelContainer: {
		position: 'absolute',
		top: 0,
		bottom: 0,
		left: 0,
		right: 0,
		flex: 1,
	},
	viewContainer: {
		flex: 1,
	},
	panel: {
		width: Screen.width,
		height: Screen.height,
	},

	panelVisible: {
		paddingTop: 20,
		width:"100%",
		height: Screen.height,
		backgroundColor: backgroundColor,
		borderTopLeftRadius: 20,
		borderTopRightRadius: 20,
		shadowColor: '#000000',
		shadowOffset: {width: 0, height: 0},
		shadowRadius: 5,
		shadowOpacity: 0.4
	},

	panelHeader: {
		alignItems: 'center',
		position: 'relative',
		height: 20
	},
	panelHandle: {
		width: 40,
		height: 8,
		borderRadius: 4,
		backgroundColor: detailColor,
		top: 30,
		zIndex: 1,
		position: "absolute"
	},
	badgeContainer: {
		backgroundColor: '#800000',
		position: 'relative',
		top: 22,
		left: 50
	},
	badgeText: {
		color: '#FFFFFF'
	}
})

const mapStateToProps = (state) => {
	return {
		currentRegion: state.map.currentRegion,
		currentLocation: state.map.location,
		markerList: state.markers.markerList,
		markerViewVisible: state.ui.markerView.markerViewVisible
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		menuButtonPress: () => {
			dispatch(Actions.drawerOpen())
		},
		regionChange: (region) => {
			dispatch(Actions.updateRegion(region))
		},
		setMarkerSelected: (marker) => {
			dispatch(Actions.setMarkerSelected(marker))
		},
		setMarkerViewVisible: () => {
			dispatch(Actions.setMarkerViewVisible())
		},
		setMarkerViewHidden: () => {
			dispatch(Actions.setMarkerViewHidden())
		},
		disableGestures: (value) => {
			dispatch(Actions.disableGestures(value))
		}
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(MapContainer)
