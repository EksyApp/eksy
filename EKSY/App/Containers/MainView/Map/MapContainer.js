import React, {Component} from 'react'
import * as ReduxActions from "../../../Actions/index";
import MapComponent from "../../../Components/MainView/Map/MapComponent";
import {connect} from "react-redux";
import PropTypes from 'prop-types'
import {LocationShape, MarkersShape, RegionShape} from "../../../Utils/PropTypeShapes";

//Renders the map and handles it's logic
export class MapContainer extends Component {

	render() {
		return (
				<MapComponent
						currentRegion={this.props.currentRegion}
						regionChange={this.props.regionChange}
						markerList={this.props.markerList}
						setMarkerSelected={this.props.setMarkerSelected}
						setMarkerViewVisible={this.props.setMarkerViewVisible}
						disableGestures={this.props.disableGestures}
						currentLocation={this.props.currentLocation}
						radius={this.props.radius}
				/>
		)
	}

}

MapContainer.propTypes = {
	currentRegion: RegionShape,
	regionChange: PropTypes.func,
	markerList: MarkersShape,
	setMarkerSelected: PropTypes.func,
	setMarkerViewVisible: PropTypes.func,
	disableGestures: PropTypes.func,
	currentLocation: LocationShape,
	radius: PropTypes.number
}

const mapStateToProps = (state) => {
	return {
		currentRegion: state.map.currentRegion,
		currentLocation: state.map.location,
		markerList: state.markers.markerList,
		radius: state.map.radius
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		regionChange: (region) => {
			dispatch(ReduxActions.updateRegion(region))
		},
		setMarkerSelected: (marker) => {
			dispatch(ReduxActions.setMarkerSelected(marker))
		},
		setMarkerViewVisible: () => {
			dispatch(ReduxActions.setMarkerViewVisible())
		},
		disableGestures: (value) => {
			dispatch(ReduxActions.disableGestures(value))
		}
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(MapContainer)
