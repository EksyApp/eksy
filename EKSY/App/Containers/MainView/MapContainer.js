import React, {Component} from 'react'
import * as ReduxActions from "../../Actions/";
import MapComponent from "../../Components/MainView/MapComponent";
import {connect} from "react-redux";

export class MapContainer extends Component {
	
	render() {
		return (
				<MapComponent
						currentRegion={this.props.currentRegion}
						regionChange={this.props.regionChange}
						markerList={this.props.markerList}
						setMarkerSelected={this.props.setMarkerSelected}
						setMarkerViewVisible={this.props.setMarkerViewVisible}
						disableGestures={this.props.disableGestures}
						currentLocation={this.props.currentLocation}
				/>
		)
	}
	
}

const mapStateToProps = (state) => {
	return {
		currentRegion: state.map.currentRegion,
		currentLocation: state.map.location,
		markerList: state.markers.markerList,
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		regionChange: (region) => {
			dispatch(ReduxActions.updateRegion(region))
		},
		setMarkerSelected: (marker) => {
			dispatch(ReduxActions.setMarkerSelected(marker))
		},
		setMarkerViewVisible: () => {
			dispatch(ReduxActions.setMarkerViewVisible())
		},
		disableGestures: (value) => {
			dispatch(ReduxActions.disableGestures(value))
		}
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(MapContainer)