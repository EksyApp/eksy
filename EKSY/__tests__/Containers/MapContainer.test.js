import React from 'react'
import { MapContainer } from '../../App/Containers/MainView/Map/MapContainer'
import renderer from 'react-test-renderer'

test('MapContainer renders correctly', () => {
	const tree = renderer.create(
			<MapContainer
					markerList = {[{latitude: 1, longitude: 1, title: "", text: ""}]}
					currentRegion={{latitude: 1, longitude: 1}}
					regionChange={() => {}}
					setMarkerSelected={() => {}}
					setMarkerViewVisible={() => {}}
					disableGestures={() => {}}
					currentLocation={{latitude: 1, longitude: 1, isKnown: true}}
					radius={10}
					routeIsActive
					route={{title: "", text: "", markers: [{latitude: 1, longitude: 1, title: "", text: ""}]}}
					nextMarker = {{latitude: 2, longitude: 1, title: "", text: ""}}
					setNextMarker={() => {}}
					setRouteIsActive={() => {}}
			/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});