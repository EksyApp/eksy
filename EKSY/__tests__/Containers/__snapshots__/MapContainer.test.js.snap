// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapContainer renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-end",
    }
  }
>
  <MapView
    followUserLocation={true}
    initialRegion={
      Object {
        "latitude": 1,
        "longitude": 1,
      }
    }
    onRegionChange={[Function]}
    showCompass={false}
    showsBuildings={true}
    showsMyLocationButton={true}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View>
      <View>
        <Circle
          center={
            Object {
              "isKnown": true,
              "latitude": 1,
              "longitude": 1,
            }
          }
          fillColor="rgba(66, 180, 230, 0.2)"
          radius={10000}
          strokeColor="rgba(66, 180, 230, 1)"
          strokeWidth={0.5}
        />
        <Circle
          center={
            Object {
              "isKnown": true,
              "latitude": 1,
              "longitude": 1,
            }
          }
          fillColor="rgba(66, 180, 230, 1)"
          radius={NaN}
          strokeColor="rgba(66, 180, 230, 1)"
          strokeWidth={0.5}
        />
      </View>
      <View>
        <View />
        <Marker
          coordinate={
            Object {
              "latitude": 1.089932160591873,
              "longitude": 1,
            }
          }
          onPress={[Function]}
          pinColor="rgb(211,211,211)"
        />
        <Polyline
          coordinates={Array []}
        />
      </View>
    </View>
  </MapView>
</View>
`;
